src/modeler/AutoRest.Swagger/Validation/ServiceDefinitionParameters.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using AutoRest.Core.Logging;
+using AutoRest.Core.Properties;
+using AutoRest.Core.Validation;
+using AutoRest.Swagger.Model;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class ServiceDefinitionParameters : TypedRule<Dictionary<string, SwaggerParameter>>
+    {
+        private const string SubscriptionId = "subscriptionId";
+        private const string ApiVersion = "api-version";
+        /// <summary>
+        /// This rule passes if the parameters section contains both subscriptionId and api-version parameters
+        /// </summary>
+        /// <param name="paths"></param>
+        /// <returns></returns>
+        public override bool IsValid(Dictionary<string, SwaggerParameter> ParametersMap)
+        {
+            if (ParametersMap.Count==0 || ParametersMap == null)
+                return false;
+
+            var Names = ParametersMap.Values.Select(Param => Param.Name).ToList();
+            return Names.Contains(SubscriptionId) && Names.Contains(ApiVersion);
```

Comment:
The code 
``` csharp 
var Names = ParametersMap.Values.Select(Param => Param.Name).ToList();
return Names.Contains(SubscriptionId) && Names.Contains(ApiVersion);
```
is iterating the `ParametersMap.Values` enumerable and allocating a `List` for no benefit. 
`.ToList()` should be reserved for when you actually need to manipulate the result set of an enumeration.

The following will iterate the set only once, and will return at the first `true` case. If there are a 1000 items in the list, and the first is matched, the iterating stops there. The happy path is the fast path ⏩ 
``` csharp
return ParametersMap.Values.Any( parameter => parameter.Name == SubscriptionId  || parameter.Name == ApiVersion );
```